### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by ericgaspar**
- upgrade to 25.0.1

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by MCMic** [comment]
Note that 25 breaks 32bit support in its current state.
Also most applications are not released yet for 25.

**Comment by alexAubin** [comment]
Ogod, Nextcloud is indeed dropping support for 32 bit ... This is going to be fun :/

https://help.nextcloud.com/t/32-bit-support-is-no-longer-available-from-nextcloud-what-are-the-takeaways/143879

**Comment by lapineige** [comment]
One update went fine… but one other (server) no longer allows me to connect :(

**Comment by lapineige** [comment]
> Ogod, Nextcloud is indeed dropping support for 32 bit ... This is going to be fun :/
> 
> https://help.nextcloud.com/t/32-bit-support-is-no-longer-available-from-nextcloud-what-are-the-takeaways/143879

Could this upgrade be disabled on all 32-bits systems ?

**Comment by alexAubin** [comment]
>Could this upgrade be disabled on all 32-bits systems ?

We can probably check at the beginning of the script if the system is 32 bits and immediately stop the upgrade process, that's pretty easy

What I'm worried about is that "we" (the YunoHost project) probably have quite a lot of users who are running a 32 bit server (eg a RPi 2, or even RPi 3 or 4 if running with the 'old' 32 bit OS) and people ultimately have the choice between "buying a new hardware / reinstalling from scratch" (= super annoying and bad for ecology etc.) OR "keep using Nextcloud 24 for X years before ultimately buying a new hardware / reinstalling from scratch)" (= bad for security and still pretty annoying) :worried:

**Comment by lapineige** [comment]
> One update went fine… but one other (server) no longer allows me to connect :(

Ok I think this issue was unrelated and is now gone.

But on that server, Nextcloud, on the `/settings/admin/overview` page, I have this message:
>     It seems like you are running a 32-bit PHP version. Nextcloud needs 64-bit to run well. Please upgrade your OS and PHP to 64-bit!

It's a Raspberry Pi 4. Nextcloud is running well as far as I can tell.
What's happening ? :thinking:

**Comment by lapineige** [comment]
By the way, until Yunohost Nextcloud package provide 24.0.6, this is the only branch that help fix the last disclosed (quite big) security issues : https://www.cert.ssi.gouv.fr/avis/CERTFR-2022-AVI-961/

**Comment by alexAubin** [comment]
>>It seems like you are running a 32-bit PHP version. Nextcloud needs 64-bit to run well. Please upgrade your OS and PHP to 64-bit!
>
>It's a Raspberry Pi 4. Nextcloud is running well as far as I can tell.
>What's happening ?

I didn't really digged in to have specific details, I think overall they are just now focusing on the 64-bit use case, and 32 bit installation may work but have lower performances because they're won't be trying to keep it "smooth and optimized" anymore ...

Imho this is mainly a depreciation message considering they'll drop support for it ... which imho should have been made like 1 year ago and not just right before they drop support in version 25 ? x_X

**Comment by Sebseb01** [comment]
Hello, 

I have upgraded my Nextcloud 22 to 25 (with testing branch) without any dificulty. I have an yunohost in an lxc container on a X86_64 (Bullseye).

Some apps are not ready for 25. But migration process works well.

Thanks

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Review by tituspijean — DISMISSED** [review]

File: scripts/install [hunk 1]
```diff
@@ -40,6 +40,12 @@ test ! -e "$final_path" || ynh_die --message="This path already contains a folde
 # Register (book) web path
 ynh_webpath_register --app=$app --domain=$domain --path_url=$path_url
 
+# Check machine architecture (Seens 25.0 Nextcloud doesn't support 32bit machines)
+if [ $YNH_ARCH == "i386" ];
```
Comment (alexAubin) [inline scripts/install:44]: NB we shall probably also stop there for armhf too :/ ...

File: scripts/upgrade [hunk 2]
```diff
@@ -34,6 +34,12 @@ fpm_usage=$(ynh_app_setting_get --app=$app --key=fpm_usage)
 
 upgrade_type=$(ynh_check_app_version_changed)
 
+# Check machine architecture (Seens 25.0 Nextcloud doesn't support 32bit machines)
+if [ $YNH_ARCH == "i386" ];
```
Comment (alexAubin) [inline scripts/upgrade:38]: (same here)

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by nathanael-h** [comment]
!testme

**Comment by lapineige** [comment]
Let's merge ?

**Comment by lapineige** [comment]
> Sorry, Nextcloud has deprecated 32-bit support

This has to be announced somewhere, especially for Raspberry Pi owners…

**Review by tituspijean — APPROVED** [review]

**Comment by lapineige** [comment]
> > > It seems like you are running a 32-bit PHP version. Nextcloud needs 64-bit to run well. Please upgrade your OS and PHP to 64-bit!
> > 
> > 
> > It's a Raspberry Pi 4. Nextcloud is running well as far as I can tell.
> > What's happening ?
> 
> I didn't really digged in to have specific details, I think overall they are just now focusing on the 64-bit use case, and 32 bit installation may work but have lower performances because they're won't be trying to keep it "smooth and optimized" anymore ...
> 
> Imho this is mainly a depreciation message considering they'll drop support for it ... which imho should have been made like 1 year ago and not just right before they drop support in version 25 ? x_X

End of support for 32bits in Nextcloud 26: https://help.nextcloud.com/t/deprecation-of-32-bit-support-ends-completely-with-nextcloud-26/150926

**Comment by lapineige** [comment]
So now I understand : 32 bit is still supported, but until this 25.0.2 version there were a bug and it wasn't working: https://github.com/nextcloud/server/pull/34905

So let's remove the blocking warning, and just warn people this will be the last version (v25) available and they need to switch to 64 bit, but don't block them ?

**Comment by lapineige** [comment]
PR here: https://github.com/YunoHost-Apps/nextcloud_ynh/pull/531

**Pull request merged**